gradio/components.py

```diff
@@ -3993,9 +3990,24 @@ def update(
         }
         return updated_config
 
+    def _process_chat_messages(self, chat_message: str | Dict | None):
+        if isinstance(chat_message, str) and os.path.exists(chat_message):
+            file_path = self.make_temp_copy_if_needed(chat_message)
+            mime_type = processing_utils.get_mimetype(file_path)
+            return {
+                "name": file_path,
+                "data": None,
+                "is_file": True,
+                "mime_type": mime_type,
+            }
+        elif isinstance(chat_message, dict) or chat_message is None:
+            return chat_message
+        else:
+            return self.md.renderInline(chat_message)
+
     def postprocess(
-        self, y: List[Tuple[str | None, str | None]]
-    ) -> List[Tuple[str | None, str | None]]:
+        self, y: List[Tuple[str | None | Dict, str | None | Dict]]
+    ) -> List[Tuple[str | Dict | None, str | Dict | None]]:
         """
         Parameters:
             y: List of tuples representing the message and response pairs. Each message and response should be a string, which may be in Markdown format.
```

Comment:
```suggestion
            y: List of tuples representing the message and response pairs. Each message and response should be a string, which may be in Markdown format.  It can also be a string filepath or URL to an image/video/audio in which case the file is displayed. It can also be None, in which case that message is displayed.
```